### PR TITLE
VideoPress: re-write video chapters block in TS

### PR DESCRIPTION
**Pull request by retrofox**
This is a janitorial PR:
* Re-write part of the Video Chapters block code in TypeScript

image

Screen Shot 2022-12-18 at 12 33 33

* Add missing `@wordpress/dom-ready` dependency
* Remove unused editor.js file

Part of https://github.com/Automattic/jetpack/issues/27867

#### Changes proposed in this Pull Request:

* VideoPress: re-write video chapters block in TS

#### Testing instructions:

* Enable beta extensions
* Go to block editor
* Insert the Video Chapters block
* Confirm it should work as expected. No significant changes here.

image

### CODE REVIEW

File: projects/packages/videopress/src/client/block-editor/blocks/video-chapters/save.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+ * Types
+ */
+import type React from 'react';
+
+/**
+ * VideoPress block save function
+ *
+ * @returns {React.ReactElement}    Component template
+ */
+export default function save(): React.ReactElement {
```
Comment (renatoagds) [inline projects/packages/videopress/src/client/block-editor/blocks/video-chapters/save.ts]:
```suggestion
export default function save(): React.ReactNode {
```

File: projects/packages/videopress/src/client/block-editor/blocks/video-chapters/save.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+/**
+ * Types
+ */
+import type React from 'react';
+
+/**
+ * VideoPress block save function
+ *
+ * @returns {React.ReactElement}    Component template
```
Comment (renatoagds) [inline projects/packages/videopress/src/client/block-editor/blocks/video-chapters/save.ts]:
```suggestion
 * @returns {React.ReactNode}       Component template
```

File: projects/packages/videopress/src/client/block-editor/blocks/video-chapters/edit.tsx [hunk 3]
```diff
@@ -12,23 +9,22 @@ import classNames from 'classnames';
...
  * VideoPress Chapters block Edit react components
  *
- * @param {object} props                 - Component props.
- * @param {object} props.attributes      - Block attributes.
- * @param {Function} props.setAttributes - Function to set block attributes.
- * @param {boolean} props.isSelected     - Whether the block is selected.
- * @param {string} props.clientId        - Block client ID.
- * @returns {object}                     - React component.
+ * @param {VideoChaptersBlockPropertiesProps} props - Component props.
+ * @returns {React.ReactElement}                      React component.
```
Comment (renatoagds) [inline projects/packages/videopress/src/client/block-editor/blocks/video-chapters/edit.tsx]:
```suggestion
 * @returns {React.ReactNode}                         React component.
```

**Comment by retrofox** [comment]
Thanks @renatoagds for your review 🙇‍♂️ Going to update the PT soon.

**Pull request closed without merging**